src/services/goals.js

```diff
@@ -377,8 +377,13 @@ export function reduceObjectives(newObjectives, currentObjectives = []) {
 
 export function reduceObjectivesForActivityReport(newObjectives, currentObjectives = []) {
   return newObjectives.reduce((objectives, objective) => {
+    const objectiveStatus = objective.activityReportObjectives
```

Comment:
As we discussed might not be bad idea to either add comment or rename 'objectives' to 'accumObjectives'.

Also duplicate code we saw below.